File: SECO2425C073 (3rd Sem)/OOP/Practical5.cpp
#include <iostream>
#include <vector>
using namespace std;

template <typename T>
void sort(T arr[], int n) 
{
    int min;
    for (int i = 0; i < n - 1; i++) 
    {
        min = i;
        for (int j = i + 1; j < n; j++) 
        {
            if (arr[min] > arr[j]) 
            { 
                min = j;
            }
        }
        // Swap elements
        T temp = arr[i];
        arr[i] = arr[min];
        arr[min] = temp;
    }
    // Output the sorted array
    for (int i = 0; i < n; i++) 
    {
        cout << arr[i] << " ";
    }
    cout << endl;
}

int main() {
    int ch, n;
    do 
    {
        cout << "\n\nMenu" << endl;
        cout << "1. Sorting of int array" << endl;
        cout << "2. Sorting of float array" << endl;
        cout << "3. Exit" << endl;
        cout << "Enter Your Choice: ";
        cin >> ch;
        
        if (ch == 3) {
            cout << "Exiting..." << endl;
            break;
        }

        cout << "Enter Size of array: ";
        cin >> n;

        if (ch == 1) 
        {
            vector<int> arr(n); // Use vector for dynamic sizing
            cout << "Enter " << n << " integers:" << endl;
            for (int i = 0; i < n; i++) 
            {
                cin >> arr[i];
            }
            sort<int>(arr.data(), n);  // Pass array to sort function
        } 
        else if (ch == 2) 
        {
            vector<float> arr2(n); // Use vector for dynamic sizing
            cout << "Enter " << n << " floats:" << endl;
            for (int i = 0; i < n; i++) 
            {
                cin >> arr2[i];
            }
            sort<float>(arr2.data(), n);  // Pass array to sort function
        } 
        else {
            cout << "Invalid choice. Please select 1, 2, or 3." << endl;
        }

    } while (ch != 3);

    return 0;
}

File: SECO2425C073 (3rd Sem)/DSL/Practical_5.py
def menu():
    arr = [30, 20, 1, 5, 34,90]
    ch = 1
    while ch != 4:
        print("################Menu#################")
        print("1. Bubble Sort")
        print("2. Selection Sort")
        print("3.Top 5 scores")
        print("4. Exit")
        ch = int(input("Enter your choice: "))
        if ch == 1:
            bubble_sort(arr)
        elif ch == 2:
            selection_sort(arr)
        elif ch==3:
            print(arr[0:6])
        elif ch==4:
            break
        else:
            print("Invalid Option")

def bubble_sort(arr):
    for i in range(0, len(arr) - 1):
        for j in range(0, len(arr) - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    print("Sorted array (Bubble Sort):", arr)

def selection_sort(arr):
    for i in range(len(arr) - 1):
        min_index = i
        for j in range(i + 1, len(arr)):
            if arr[j] < arr[min_index]:
                min_index = j
        arr[i], arr[min_index] = arr[min_index], arr[i]
    print("Sorted array (Selection Sort):", arr)

menu()

